src/spec/expectations.cr

```diff
@@ -50,109 +115,139 @@ module Spec
       end
     end
 
-    def negative_failure_message(actual_value)
+    def negative_failure_message(actual_value) : String
       "Expected: actual_value != #{@expected_value.inspect}\n     got: #{actual_value.inspect}"
     end
   end
 
   # :nodoc:
   struct BeExpectation(T)
+    include Expectation
+
     def initialize(@expected_value : T)
     end
 
-    def match(actual_value)
+    def match(actual_value) : Bool
       actual_value.same? @expected_value
     end
 
-    def failure_message(actual_value)
+    def failure_message(actual_value) : String
       "Expected: #{@expected_value.inspect} (object_id: #{@expected_value.object_id})\n     got: #{actual_value.inspect} (object_id: #{actual_value.object_id})"
     end
 
-    def negative_failure_message(actual_value)
+    def negative_failure_message(actual_value) : String
       "Expected: value.same? #{@expected_value.inspect} (object_id: #{@expected_value.object_id})\n     got: #{actual_value.inspect} (object_id: #{actual_value.object_id})"
     end
   end
 
   # :nodoc:
   struct BeTruthyExpectation
-    def match(actual_value)
+    include Expectation
+
+    def match(actual_value) : Bool
       !!actual_value
     end
 
-    def failure_message(actual_value)
+    def failure_message(actual_value) : String
       "Expected: #{actual_value.inspect} to be truthy"
     end
 
-    def negative_failure_message(actual_value)
+    def negative_failure_message(actual_value) : String
       "Expected: #{actual_value.inspect} not to be truthy"
     end
   end
 
   # :nodoc:
   struct BeFalseyExpectation
-    def match(actual_value)
+    include Expectation
+
+    def match(actual_value) : Bool
       !actual_value
     end
 
-    def failure_message(actual_value)
+    def failure_message(actual_value) : String
       "Expected: #{actual_value.inspect} to be falsey"
     end
 
-    def negative_failure_message(actual_value)
+    def negative_failure_message(actual_value) : String
       "Expected: #{actual_value.inspect} not to be falsey"
     end
   end
 
   # :nodoc:
   struct BeNilExpectation
-    def match(actual_value)
+    include Expectation
+
+    def match(actual_value) : Bool
       actual_value.nil?
     end
 
-    def failure_message(actual_value)
+    def failure_message(actual_value) : String
       "Expected: #{actual_value.inspect} to be nil"
     end
 
-    def negative_failure_message(actual_value)
+    def negative_failure_message(actual_value) : String
       "Expected: #{actual_value.inspect} not to be nil"
     end
+
+    def cast_should(value)
+      value.nil? ? value : (raise "expected #{value} to be nil")
```

Comment:
These should be `BUG: `right?

Reply:
Well, you can always instantiate this and call the method and you'll get an exception. I use BUG for compiler bugs.